plugins/com.google.cloud.tools.eclipse.appengine.standard.java8/src-test/com/google/cloud/tools/eclipse/appengine/standard/java8/AppEngineXsltTransformTests.java

```diff
@@ -0,0 +1,136 @@
+        "<?xml version=\"1.0\" encoding=\"UTF-8\"?><ae:appengine-web-app xmlns:ae=\"http://appengine.google.com/ns/1.0\">"
+            + "<runtime xmlns=\"http://appengine.google.com/ns/1.0\">java8</runtime>"
+            + "</ae:appengine-web-app>");
+  }
+
+  @Test
+  public void testRemoveBare() throws IOException, TransformerException {
+    apply("/xslt/removeJava8Runtime.xsl",
+        "<?xml version=\"1.0\" encoding=\"UTF-8\"?><appengine-web-app xmlns=\"http://appengine.google.com/ns/1.0\"/>",
+        "<?xml version=\"1.0\" encoding=\"UTF-8\"?><appengine-web-app xmlns=\"http://appengine.google.com/ns/1.0\"/>");
+  }
+
+  @Test
+  public void testRemovePreservesNonJava8Runtime() throws IOException, TransformerException {
+    apply("/xslt/removeJava8Runtime.xsl",
+        "<?xml version=\"1.0\" encoding=\"UTF-8\"?><appengine-web-app xmlns=\"http://appengine.google.com/ns/1.0\">"
+            + "<runtime>XXX</runtime></appengine-web-app>",
+        "<?xml version=\"1.0\" encoding=\"UTF-8\"?><appengine-web-app xmlns=\"http://appengine.google.com/ns/1.0\">"
+            + "<runtime>XXX</runtime></appengine-web-app>");
+  }
+
+  @Test
+  public void testRemovePreservesOrder() throws IOException, TransformerException {
+    apply("/xslt/removeJava8Runtime.xsl",
+        "<?xml version=\"1.0\" encoding=\"UTF-8\"?><appengine-web-app xmlns=\"http://appengine.google.com/ns/1.0\">"
+            + "<threadsafe>true</threadsafe>" + "<runtime>java8</runtime>"
+            + "<sessions-enabled>true</sessions-enabled>" + "</appengine-web-app>",
+        "<?xml version=\"1.0\" encoding=\"UTF-8\"?><appengine-web-app xmlns=\"http://appengine.google.com/ns/1.0\">"
+            + "<threadsafe>true</threadsafe>" + "<sessions-enabled>true</sessions-enabled>"
+            + "</appengine-web-app>");
+  }
+
+  @Test
+  public void testRemoveExistingJava8Runtime() throws IOException, TransformerException {
+    apply("/xslt/removeJava8Runtime.xsl",
+        "<?xml version=\"1.0\" encoding=\"UTF-8\"?><appengine-web-app xmlns=\"http://appengine.google.com/ns/1.0\">"
+            + "<runtime>java8</runtime></appengine-web-app>",
+        "<?xml version=\"1.0\" encoding=\"UTF-8\"?><appengine-web-app xmlns=\"http://appengine.google.com/ns/1.0\"/>");
+  }
+
+  private void apply(String templateFile, String inputString, String outputString)
+      throws IOException, TransformerException {
+    URL xslPath = AppEngineXsltTransformTests.class.getResource(templateFile);
+    if (xslPath == null) {
+      File xslFile = new File("." + templateFile);
+      assertTrue(xslFile.exists());
+      xslPath = xslFile.toURI().toURL();
+    }
+    InputStream documentStream =
+        new ByteArrayInputStream(inputString.getBytes(StandardCharsets.UTF_8));
+    InputStream transformed = Xslt.applyXslt(documentStream, xslPath.openStream());
+    String result =
```

Comment:
You need to feed the string returned into a parser, then query the resulting DOM object to see if what you want/don't want is/isn't present. Some of Abbie's tests do this already, I think.